libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteDetector.kt

```diff
@@ -3,17 +3,18 @@ package com.mapbox.navigation.core.fasterroute
 import com.mapbox.api.directions.v5.models.DirectionsRoute
 import com.mapbox.navigation.base.trip.model.RouteProgress
 
-internal object FasterRouteDetector {
-
-    /**
-     * This threshold optimizes for the current route. For example, if you're 40 minutes
-     * away and there is an alternative that is 1 minute faster, it will be ignored.
-     */
-    private const val PERCENTAGE_THRESHOLD = 0.90
-
+internal class FasterRouteDetector {
```

Comment:
Why do we change `FasterRouteDetector` back into a `class`? IMO `FasterRouteDetector#isRouteFaster` is a pure function so it makes sense to be an `object`. We're not injecting `FasterRouteDetector` either https://github.com/mapbox/mapbox-navigation-android/blob/168bb4e17eac1c4183f1f9ae269322886629e6f1/libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteController.kt#L17 On a side note, we're not testing https://github.com/mapbox/mapbox-navigation-android/blob/168bb4e17eac1c4183f1f9ae269322886629e6f1/libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteController.kt#L52-L56 To do so we can either revert these changes and use MockK `mockkObject` / `unmockkObject` or inject `FasterRouteDetector` into `FasterRouteController`. What do you think?

Reply:
> Why do we change FasterRouteDetector back into a class? MO FasterRouteDetector#isRouteFaster is a pure function so it makes sense to be an object.

https://kotlinlang.org/docs/reference/object-declarations.html
https://kotlinlang.org/docs/tutorials/kotlin-for-py/objects-and-companion-objects.html

For the most part, kotlin documentation suggests an object's purpose is for creating Singletons. Agree they can be used for computation such as `Math`, but kotlin [doesn't even do Math that way anymore](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/jvm/src/kotlin/util/MathJVM.kt). 

> On a side note, we're not testing .. To do so we can either revert these changes and use MockK mockkObject / unmockkObject or inject FasterRouteDetector into FasterRouteController. What do you think?

The tests behave the same whether it is an object or a class. But I definitely agree there should be another test that demonstrates "onFasterRouteAvailable` is being called. Adding it ❤️

Reply:
it's also reasonable to say the class could be deleted and moved into the controller. But then again, only fixing a bug here 😄